backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/product/ProductRestService.java

```diff
@@ -363,6 +363,12 @@ private void createOrUpdateBpartnerProduct(
 			if (effectiveSyncAdvise.getIfExists().isUpdate())
 			{
 				final BPartnerProduct bPartnerProduct = syncBPartnerProductWithJson(jsonRequestBPartnerProductUpsert, existingBPartnerProduct, bPartnerId);
+
+				if (Boolean.TRUE.equals(bPartnerProduct.getCurrentVendor()))
+				{
+					productRepository.resetCurrentVendorFor(productId);
```

Comment:
afais this reset now also happens even if the if `bPartnerProduct.getCurrentVendor() == true` already. please du the update only if needed